helpers.go

```diff
@@ -70,3 +73,118 @@ func getListOfVersionedRepositories(inVersion string) ([]string, error) {
 
 	return strings.Split(strings.TrimSpace(string(output)), "\n"), nil
 }
+
+func validDependabotPR(pr *github.PullRequestEvent) bool {
+	// NOTE: For now, checking the 'labeled' condition, and not 'opened' (for testing purposes)
+	return pr.GetAction() == "labeled" && pr.GetSender().GetLogin() == "dependabot[bot]"
+}
+
+func maybeVendorDependabotPR(pr *github.PullRequestEvent) error {
+	if !validDependabotPR(pr) {
+		// Just ignore
+		log.Debug("Not a valid dependabot PR, ignoring...\n")
+		log.Debugf("Action: %s\n", pr.GetAction())
+		log.Debugf("Name: %v\n", pr.GetSender().GetLogin())
+		return nil
+	}
+
+	head := pr.GetPullRequest().GetHead().GetRef()
+	if head == "" {
+		return fmt.Errorf("No Head set in the PR: %v", pr)
+	}
+
+	repo := pr.GetRepo().GetName()
+	if repo == "" {
+		return fmt.Errorf("No repository name found in PR: %v", pr)
+	}
+
+	nr := pr.GetPullRequest().GetNumber()
+	if nr == 0 {
+		return fmt.Errorf("No PR number found in PR: %v", pr)
+	}
+
+	tmpdir, err := ioutil.TempDir("", repo)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	cmd := exec.Command("git", "init", ".")
+	cmd.Dir = tmpdir
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", cmd.Args, out, err.Error())
+	}
+
+	log.Debugf("Adding remote: %s", repo)
+	cmd = exec.Command("git", "remote", "add", "mendersoftware", "git@github.com:mendersoftware/"+repo+".git")
+	cmd.Dir = tmpdir
+	out, err = cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", cmd.Args, out, err.Error())
+	}
+
+	localBranch := fmt.Sprintf("dependabot-%d", nr)
+	cmd = exec.Command("git", "fetch", "mendersoftware", fmt.Sprintf("%s:%s", head, localBranch))
+	cmd.Dir = tmpdir
+	out, err = cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", cmd.Args, out, err.Error())
+	}
+
+	log.Debug("Checking out the local branch..")
+	cmd = exec.Command("git", "checkout", "--quiet", localBranch)
+	cmd.Dir = tmpdir
+	out, err = cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s: %s", cmd.Args, out, err.Error())
+	}
```

Comment:
If `head` is just a regular branch name, we can simplify all these commands with just something like `"git clone -b "+head+ "git@github.com:mendersoftware/"+repo+".git"`

Reply:
I don't know what this magic is, but sure :smile_cat:

Reply:
Should we also add the `--single-branch` option then, for clone economy?

Reply:
@oleorhagen Good idea